clients/ts/signalr/src/HubConnection.ts

```diff
@@ -508,25 +522,67 @@ export class HubConnection {
                 type: MessageType.Invocation,
             };
         } else {
-            const id = this.id;
-            this.id++;
+            const invocationId = this.invocationId;
+            this.invocationId++;
 
             return {
                 arguments: args,
-                invocationId: id.toString(),
+                invocationId: invocationId.toString(),
                 target: methodName,
                 type: MessageType.Invocation,
             };
         }
     }
 
+    private launchStreams(streams: Array<IStreamResult<any>>): void {
+        if (streams.length === 0) {
+            return;
+        }
+
+        // We want to iterate over the keys, since the keys are the stream ids
+        // tslint:disable-next-line:forin
+        for (const streamId in streams) {
+            streams[streamId].subscribe({
+                complete: async () => {
+                    await this.sendWithProtocol(this.createStreamCompleteMessage(streamId));
+                },
+                error: async (err) => {
+                    await this.sendWithProtocol(this.createStreamCompleteMessage(streamId, err));
+                },
+                next: async (item) => {
+                    await this.sendWithProtocol(this.createStreamDataMessage(streamId, item));
+                },
+            });
+        }
+    }
+
+    private replaceStreamingParams(args: any[]): Array<IStreamResult<any>> {
+        const streams: Array<IStreamResult<any>> = [];
+        for (let i = 0; i < args.length; i++) {
+            const argument = args[i];
+            if (this.isObservable(argument)) {
+                const streamId = this.nextStreamId();
+                // Store the stream for later use
+                streams[streamId] = argument;
+                // Replace the stream with a placeholder
+                args[i] = { streamId };
+            }
+        }
+
+        return streams;
+    }
+
+    private isObservable(arg: any): arg is IStreamResult<any> {
+        return arg.subscribe && typeof arg.subscribe === "function";
```

Comment:
nit: This seems fragile.

Reply:
Welcome to TS/JS

Reply:
Yeah, this is totally the legit JS was to do this. It's not pretty, but it works :)